lib/api/src/sys/function_env.rs

```diff
@@ -117,6 +117,17 @@ impl<T: Send + 'static> FunctionEnvMut<'_, T> {
             func_env: self.func_env.clone(),
         }
     }
+
+    /// Borrows a new mutable reference of both the attached Store and host state
+    pub fn data_and_store_mut(&mut self) -> (&mut T, StoreMut) {
+        let data = self.func_env.as_mut(&mut self.store_mut) as *mut T;
+        // telling the borrow check to close his eyes here
```

Comment:
After reviewing the API, I don't see any other (public) way  to get access to the `FunctionEnv` in `StoreInner`, so this should be safe from a public API standpoint. 

Of course internally we could still violate this. 

A thing I thought about yesterday : we could put the FunctionEnvs behind a `RefCell` to make this a bit more ssafer, but that would come with  some overhead.

But otherwise I think this is a not ideal, but acceptable use of unsafe.

Reply:
Yeah, agreed